pages/components/invoice-card.tsx

```diff
@@ -54,22 +66,21 @@ export default function CheckoutCard({
       });
   }, []);
 
-  useEffect(() => {
-    if (randomNsec !== "") {
-      handlePayment(totalCost, currency);
-    }
-  }, [randomNsec]);
-
   useEffect(() => {
     const profileMap = profileContext.profileData;
     const profile = profileMap.has(decryptedNpub)
       ? profileMap.get(decryptedNpub)
       : undefined;
-    setName(
-      profile && profile.content.name
-        ? profile.content.name
-        : nip19.npubEncode(decryptedNpub),
-    );
+    if (typeof window !== "undefined") {
```

Comment:
Still think this check is not needed. Am free to hop on a call to discuss

Reply:
If you remove this it'll fail to load.